src/main/java/com/google/api/codegen/DiscoveryFragmentGeneratorApi.java

```diff
@@ -115,6 +126,28 @@ protected void process() throws Exception {
     }
   }
 
+  /*
+   * Parses strings of the format
+   * "go:https://www.hello.com,java:https://www.world.com" and returns the value
+   * of the key corresponding to id.
+   */
+  private String parseAuthUrl(String authUrl, String id) {
```

Comment:
So I had originally gone about implementing with a map... but since the generator operates on a language by language basis, it didn't seem worth storing the other language pairs for any reason. I figured returning when we've hit the right key was sufficient.

I hope that with the comments the function's behavior is pretty clear.